src/rust/engine/cargo_build_shim.sh

```diff
@@ -0,0 +1,13 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# This simply exists because `shell_command` at the time of writing doesn't know how to merge `PATH`
+# from the rule's env and the `extra_env_vars`.
+
+PYTHON_PATH=${CHROOT}/3rdparty/tools/python3/python/bin
+export PATH=$PATH:$PYTHON_PATH
```

Comment:
It's unlikely, but these should be quoted in case of spaces:

```suggestion
PYTHON_PATH="${CHROOT}/3rdparty/tools/python3/python/bin"
export PATH="$PATH:$PYTHON_PATH"
```